workshops/pacman game/README.md

```diff
@@ -0,0 +1,337 @@
+* We're using a for loop to traverse through the array to add the array items one by one. 
+  * If the value at a specific spot is `0`, we'll add a `brick` (from the function we coded earlier).
+  * If the value is `1`, then we'll fill it in for the Pac-man to walk through.
+* Remember to be very careful about the indentation of each line — for example, make sure the `path.up()` is lined up exactly under the `if` syntax.
+
+## Moving the Pac-man
+
+We want our Pac-man to be able to move through the world we created. We're going to create a new function called `move` that will allow the character to move. While coding, notice the different variables we're using and functions that correspond.
+
+```python
+def move():
+    writer.undo()
+    writer.write(state['score'])
+    clear()    
+    if valid(pacman + aim):
+        pacman.move(aim)
+
+    index = offset(pacman)
+
+    if tiles[index] == 1:
+        tiles[index] = 2
+        state['score'] += 1
+        x = (index % 20) * 20 - 200
+        y = 180 - (index // 20) * 20
+        brick(x, y)
+
+    up()
+    goto(pacman.x + 10, pacman.y + 10)
+    dot(20, 'yellow')
+
+    for point, course in ghosts:
+        if valid(point + course):
+            point.move(course)
+        else:
+            options = [
+                vector(5, 0),
+                vector(-5, 0),
+                vector(0, 5),
+                vector(0, -5),
+            ]
+            plan = choice(options)
+            course.x = plan.x
+            course.y = plan.y
+
+        up()
+        goto(point.x + 10, point.y + 10)
+        dot(20, 'red')
+
+    update()
+
+    for point, course in ghosts:
+        if abs(pacman - point) < 20:
+            return
+    ontimer(move, 100)
+```
+
+* **NOTE:** Our Pac-man has no sense of direction yet. It's using the default `aim` we assigned earlier when we were initializing variables. At the moment, our Pac-man is only programmed to go right, at a speed of 5. We'll add some direction in the next step.
+* The `writer` variable we implemented at the beginning is being used here with the `undo()` and `write()` functions. These are built-in functions that come with `Turtle`. 
+* As we go down further the code, we're checking if the movement is `valid`. Remember those physics functions we made? This is where we're implementing it. A move is `valid` when it doesn't involve touching a brick.
+* We're also utilizing the `ghosts` array with vectors we initialized before. This lets us have them in the game and move them around.
+* We're creating `dots` (a function from `Turtle`) that our Pac-man will eat. We are ensuring they're being created dynamically as they are attached to the Pac-man's moves. If the Pac-man doesn't move, it won't spawn more dots.
+* **Going a step further: **If you want to take things a step further and introduce an animation, sound effect, or other prettiness factor, we would introduce it after `brick(x, y)`. Try adding `print(“PAC-MAN”)` as an example.
```

Comment:
Typo on the `**` makes this render incorrectly in the HTML. the last two `**`s should be next to the `:`, not the `If`